es-ds-components/components/es-carousel.vue

```diff
@@ -0,0 +1,160 @@
+<script setup lang="ts">
+import Carousel from 'primevue/carousel';
+
+const responsiveOptions = ref([
+    {
+        // xl: 1200px
+        breakpoint: '1199px',
+        numVisible: 4,
+        numScroll: 1,
+    },
+    {
+        // lg: 992px
+        breakpoint: '991px',
+        numVisible: 3,
+        numScroll: 1,
+    },
+    {
+        // md: 768px
+        breakpoint: '767px',
+        numVisible: 2,
+        numScroll: 1,
+    },
+    {
+        // sm: 576px
+        breakpoint: '575px',
+        numVisible: 1,
+        numScroll: 1,
+    },
+]);
+
+const props = defineProps({
+    autoscroll: {
+        type: Boolean,
+        default: false,
+    },
+    dots: {
+        type: Boolean,
+        default: true,
+    },
+    items: {
+        type: Array,
+        default: () => [],
+        required: true,
+    },
+});
+
+const autoplayInterval = ref(props.autoscroll ? 3000 : 0);
+const key = ref('');
+
+const stopAutoplay = () => {
+    if (props.autoscroll) {
```

Comment:
Could this be `if(autoplayInterval.value > 0)` to ensure it will always `stopAutoPlay`?